src/vcu/src/can_handles.c

```diff
@@ -147,6 +170,69 @@ void handleButtonRequest(Frame *msg) {
   buttons.ScrollSelect = unpacked_msg.ScrollSelect;
 }
 
+void handleCurrentSensor_Power(Frame *msg) {
+  can0_CurrentSensor_Power_T unpacked_msg;
+
+  unpack_can0_CurrentSensor_Power(msg, &unpacked_msg);
+
+  cs_readings.power = unpacked_msg.result;
+}
+
+void handleMCMotor_Position_Info(Frame *msg) {
+  can0_MCMotor_Position_Info_T unpacked_msg;
+
+  unpack_can0_MCMotor_Position_Info(msg, &unpacked_msg);
+
+  mc_readings.speed = unpacked_msg.motor_speed;
+}
+
+void handleSBG_EKF_Velocity(Frame *msg) {
+  can0_SBG_EKF_Velocity_T unpacked_msg;
+
+  unpack_can0_SBG_EKF_Velocity(msg, &unpacked_msg);
+
+  imu_velocity.north = unpacked_msg.north;
+  imu_velocity.east = unpacked_msg.east;
+}
+
+void handleFrontCanNodeLeftWheelSpeed(Frame *msg) {
+  can0_FrontCanNodeLeftWheelSpeed_T unpacked_msg;
+
+  unpack_can0_FrontCanNodeLeftWheelSpeed(msg, &unpacked_msg);
+
+  wheel_speeds.front_left_32b_wheel_speed = unpacked_msg.left_32b;
+  wheel_speeds.front_left_16b_wheel_speed = unpacked_msg.left_16b;
+}
+
+void handleFrontCanNodeRightWheelSpeed(Frame *msg) {
+  can0_FrontCanNodeRightWheelSpeed_T unpacked_msg;
+
+  unpack_can0_FrontCanNodeRightWheelSpeed(msg, &unpacked_msg);
+
+  wheel_speeds.front_right_32b_wheel_speed = unpacked_msg.right_32b;
+  wheel_speeds.front_right_16b_wheel_speed = unpacked_msg.right_16b;
+}
+
+void handleMCTorque_Timer_Info(Frame *msg) {
+  can0_MCTorque_Timer_Info_T unpacked_msg;
+
+  unpack_can0_MCTorque_Timer_Info(msg, &unpacked_msg);
+
+  mc_readings.torque_feedback = unpacked_msg.torque_feedback;
+}
+
+void handleDashControls(Frame *msg) {
+  can0_DashControls_T unpacked_msg;
+
+  unpack_can0_DashControls(msg, &unpacked_msg);
+
+  control_settings.using_regen;
```

Comment:
Consider a typedef that sets Controls_Settings_T to the CANlib struct and just assign the unpacked struct to the instance declared in controls.h.